### ISSUE TITLE
Kotlin version?

### ISSUE THREAD

**Issue by happypennygames** [philburk/android-midisuite#71]
Good evening,
Do you plan on converting these to kotlin form at any point?  I've spent hours pouring over the android docs which are java centric as well as your examples, and while it seems I can get 80% of the way there with the android studio auto convert, ultimately, nothing works.
I know it's software and not hardware as both the midi pairing and midi scope apps on the playstore work just fine with my keyboard.
Thank you.

**Comment by philburk** [comment]
Good idea. We should start moving to Kotlin for these example programs.

**Comment by happypennygames** [comment]
That would be awesome!  Recognizing that this is "best effort" do you have any ballpark ETA?

**Comment by philburk** [comment]
We have no ETA. We are focussing on framework stability.

**Comment by happypennygames** [comment]
Thanks for the update!  Good news/bad news:  I was able to mine enough knowledge from your java demo to get it working for my app, so thanks for the assist!  Admittedly, it's a bit of a serious hack job, but it does work!
https://play.google.com/store/apps/details?id=com.happypennygames.aimusic.sightreader as an example.

**Comment by regalstreak** [comment]
Same here. Can't receive midi events on android from BLE. @happypennygames how did you fix the receiver?

**Comment by happypennygames** [comment]
I honestly have nothing useful to say, sorry. As described above I just
started hacking and hacking between the example files until it kind of
worked.  Apologies for the delay I kept putting off responding because I
wanted to have something more substantial to say.

**Comment by philburk** [comment]
There were recent changes to MIDI+BTLE in PR #78.
You would need to do the same things in Kotlin for Android S and above.
